viper.go

```diff
@@ -313,6 +313,8 @@ func (v *Viper) searchMap(source map[string]interface{}, path []string) interfac
 
 	if next, ok := source[path[0]]; ok {
 		switch next.(type) {
+		case map[interface{}]interface{}:
```

Comment:
Would be good to add more deep nested properties in the viper_test.go samples to be sure that such deep nested cases are always tested. Thanks!